R/make_report.R

```diff
@@ -382,18 +382,26 @@ save_svg <- function(p, output, width = 15, height = 10, units = "in", ...) {
 }
 
 
-extract_legend <- function(p) {
-  first_grob <- function(x) {
-    x$grobs[[1]]
+find_legend_grob <- function(gtbl) {
+  # Find the legend grob; tested on ggplot2 3.4.4 and 3.5.1, which have different grob layouts.
+  # On 3.4.4 it's guide-box > guides, on 3.5.1 it's guide-box-bottom > guides.
+
+  guide_box_grobs <- gtbl$grobs[grep("^guide-box", gtbl$layout$name)]
+  nonzero_grobs <- guide_box_grobs[!vapply(guide_box_grobs, inherits, logical(1), what = "zeroGrob")]
+  if (length(nonzero_grobs) != 1) {
+    stop("Couldn't find legend grob (searched for 'guide-box*')! Perhaps ggplot2's grob layout has changed?")
```

Comment:
I'd early-exit here with an empty graphic if there is no legend
```suggestion
  if (length(nonzero_grobs) == 0) {
    return(ggplot2::zeroGrob())
  }
  if (length(nonzero_grobs) > 1) {
    warn("Found legends at multiple positions. Picking the first legend.")
```